vortex-expr/src/transform/field_type.rs

```diff
@@ -0,0 +1,97 @@
+use vortex_array::{ArrayDType, Canonical, IntoArrayData};
+use vortex_dtype::DType;
+use vortex_error::{vortex_bail, VortexResult};
+
+use crate::traversal::{MutNodeVisitor, Node, TransformResult};
+use crate::{ExprRef, GetItem};
+
+pub struct FieldToNameTransform {
```

Comment:
Can you write a doc string please. And also one for `new` since it's not obvious to me what ident_dt means? It's too abbreviated